dspace-api/src/main/java/org/dspace/storage/bitstore/AzureBlobStorageBitStoreService.java

```diff
@@ -0,0 +1,395 @@
+	}
+
+	/**
+	 * Utility Method: Prefix the key with a subfolder, if this instance assets are
+	 * stored within subfolder
+	 * 
+	 * @param id
+	 * @return full key prefixed with a subfolder, if applicable
+	 */
+	private String getFullKey(String id) {
+		if (StringUtils.isNotEmpty(this.subfolder)) {
+			return this.subfolder + "/" + id;
+		} else {
+			return id;
+		}
+	}
+
+	/**
+	 * Initializes the Azure Blob Storage Service.
+	 * 
+	 * It requires: - Account Name - Account Key - Container Reference/Name
+	 */
+	@Override
+	public void init() throws IOException {
+		// check for minimum required parameters
+		if (StringUtils.isBlank(getAzureAccountName()) || StringUtils.isBlank(getAzureAccountKey())) {
+			log.warn("Empty Azure Blob Storage Account Name or Azure Account Key");
+		}
+
+		String storageConnectionString = "DefaultEndpointsProtocol=https;AccountName=".concat(getAzureAccountName())
+				.concat(";AccountKey=").concat(getAzureAccountKey());
+
+		try {
+			CloudStorageAccount azureAccount = CloudStorageAccount.parse(storageConnectionString);
+			this.serviceClient = azureAccount.createCloudBlobClient();
+
+			// container reference aka container name
+			if (StringUtils.isEmpty(getAzureContainerReference())) {
+				String defaultContainerReference = this.containerReferenceDefaultPrefix.concat(
+						DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("dspace.hostname"));
+				log.warn("Azure Container Reference/Name is not configured, setting default: "
+						+ defaultContainerReference);
+				setAzureContainerReference(defaultContainerReference);
+			}
+
+			this.blobContainer = this.serviceClient.getContainerReference(getAzureContainerReference());
+
+			if (!this.blobContainer.exists()) {
+				this.blobContainer.create();
+				log.info("Creating new Azure Container: ".concat(getAzureContainerReference()));
+			}
+		} catch (Exception e) {
+			log.error(e);
+			throw new IOException(e);
+		}
+
+		log.info("Azure Blob Storage is ready to go! Container Reference: ".concat(getAzureContainerReference())
+				.concat(System.getProperty("line.separator")));
+	}
+
+	/**
+	 * Returns an identifier unique to this asset store instance
+	 * 
+	 * @return a unique ID
+	 */
+	@Override
+	public String generateId() {
+		return Utils.generateKey();
+	}
+
+	/**
+	 * Retrieve the bits for the asset with ID. If the asset does not exist, returns
+	 * null.
+	 * 
+	 * @param bitstream
+	 *            The ID of the asset to retrieve
+	 * @exception java.io.IOException
+	 *                If a problem occurs while retrieving the bits
+	 *
+	 * @return The stream of bits, or null
+	 */
+	@Override
+	public InputStream get(Bitstream bitstream) throws IOException {
+		String key = getFullKey(bitstream.getInternalId());
+		try {
+			CloudBlob blob = this.blobContainer.getBlockBlobReference(key);
+			return (blob != null) ? blob.openInputStream() : null;
+		} catch (Exception e) {
+			log.error("get(" + key + ")", e);
```

Comment:
This `log.error` is uninformative. Should provide more information about the error occurring here, and not just provide the method name.